troubleshooting/rocksdb/src/solution/java/com/ververica/flink/training/solutions/RocksDBTuningJobOptionsFactory.java

```diff
@@ -0,0 +1,133 @@
+            LoggerFactory.getLogger(RocksDBTuningJobOptionsFactory.class);
+
+    private static final long serialVersionUID = 1L;
+
+    private static final ConfigOption<Boolean> JAVA_LOGGING =
+            key("state.backend.rocksdb.custom.javalogging")
+                    .booleanType()
+                    .defaultValue(true)
+                    .withDescription("Redirect RocksDB logging to Java (max reduce performance)");
+
+    private static final ConfigOption<Duration> DUMP_STATS_INTERVAL =
+            key("state.backend.rocksdb.custom.dumpstats")
+                    .durationType()
+                    .defaultValue(Duration.ofMinutes(5))
+                    .withDescription("If not zero, dump RocksDB stats this often (in seconds)");
+
+    private static final ConfigOption<CompressionType> COMPRESSION =
+            key("state.backend.rocksdb.custom.compression")
+                    .enumType(CompressionType.class)
+                    .noDefaultValue()
+                    .withDescription(
+                            Description.builder()
+                                    .text("Configures RocksDB compression")
+                                    .linebreak()
+                                    .text(
+                                            "For more information, please refer to %s",
+                                            link(
+                                                    "https://github.com/facebook/rocksdb/wiki/Compression"))
+                                    .build());
+
+    public static final ConfigOption<Integer> MAX_BACKGROUND_JOBS =
+            key("state.backend.rocksdb.custom.maxjobs")
+                    .intType()
+                    .noDefaultValue()
+                    .withDescription(
+                            "Specifies the maximum number of concurrent background jobs (both flushes and compactions combined). "
+                                    + "RocksDB has default configuration as '2'.");
+
+    private final DefaultConfigurableOptionsFactory defaultFactory =
+            new DefaultConfigurableOptionsFactory();
+    private ReadableConfig configuration;
+
+    @Override
+    public DBOptions createDBOptions(
+            DBOptions currentOptions, Collection<AutoCloseable> handlesToClose) {
+
+        defaultFactory.setLogLevel(InfoLogLevel.INFO_LEVEL);
+        defaultFactory.setLogFileNum(10);
+        defaultFactory.setMaxLogFileSize("10MB");
+        currentOptions = defaultFactory.createDBOptions(currentOptions, handlesToClose);
+
+        Duration statsDumpPeriod = configuration.get(DUMP_STATS_INTERVAL);
+        currentOptions.setStatsDumpPeriodSec((int) statsDumpPeriod.toMillis());
+
+        configuration
+                .getOptional(MAX_BACKGROUND_JOBS)
+                .ifPresent(currentOptions::setMaxBackgroundJobs);
+
+        if (configuration.get(JAVA_LOGGING)) {
+            Logger logger =
+                    new Logger(currentOptions) {
```

Comment:
Instead of an anonymous class, can you make this top-level with its own logger? It's weird to see rocksdb internals logged under `RocksDBTuningJobOptionsFactory` when looking through the TM. Maybe `RocksDBNativeLogger`?